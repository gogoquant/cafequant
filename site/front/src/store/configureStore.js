import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { combineReducers } from 'redux-immutable';
import { composeWithDevTools } from 'redux-devtools-extension';
import rootReducer from './reducers';

const reducer = combineReducers({
  ...rootReducer,
});

let middleware;
if (process.env.NODE_ENV === 'production') {
  middleware = applyMiddleware(thunk);
} else {
  middleware = composeWithDevTools(applyMiddleware(thunk));
}

const configureStore = initialState => {
  const store = createStore(reducer, initialState, middleware);
  if (module.hot) {
    module.hot.accept('./reducers', () => {
      // eslint-disable-next-line global-require
      const nextRootReducer = require('./reducers').default;
      store.replaceReducer(combineReducers({ ...nextRootReducer }));
    });
  }

  return store;
};
export default configureStore;
