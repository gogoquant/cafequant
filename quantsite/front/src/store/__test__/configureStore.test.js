import configureStore from '../configureStore';

describe('Redux store', () => {
  it('initialState should match snapshots', () => {
    const store = configureStore();
    expect(store.getState()).toMatchSnapshot();
  });
});
