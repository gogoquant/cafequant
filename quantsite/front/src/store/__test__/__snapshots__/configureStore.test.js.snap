// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Redux store initialState should match snapshots 1`] = `
Immutable.Map {
  "basic": Immutable.Map {
    "tab": "all",
  },
  "home": Immutable.Map {
    "loading": false,
    "loadingMore": false,
    "hasMore": true,
    "error": false,
    "topicsData": Array [],
    "scrollHeight": 0,
  },
  "login": Immutable.Map {
    "loading": false,
    "userData": Object {},
    "error": false,
  },
  "topic": Immutable.Map {
    "isCollect": false,
    "error": false,
    "upError": false,
    "loadingCollect": false,
    "repliesData": Immutable.Map {
      "entities": Immutable.Map {},
      "result": Immutable.List [],
    },
    "loadingUp": false,
    "topicData": Immutable.Map {},
    "loading": false,
    "collectError": false,
  },
  "user": Immutable.Map {
    "loading": false,
    "userData": Object {},
    "error": false,
  },
  "edit": Immutable.Map {
    "id": "",
    "tab": "dev",
    "title": "",
    "content": "",
  },
}
`;
